sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py

```diff
@@ -94,36 +82,222 @@ def _validate_job_service_type_name(self):
+        ],
+    ) -> Dict[str, RestJobService]:
         if services is None:
             return None
 
-        # pylint: disable=protected-access
         return {name: service._to_rest_object() for name, service in services.items()}
 
     @classmethod
-    def _from_rest_object(cls, obj: RestJobService) -> "JobService":
+    def _from_rest_job_service_object(cls, obj: RestJobService):
         return cls(
             endpoint=obj.endpoint,
             job_service_type=JobServiceTypeNames.REST_TO_ENTITY.get(obj.job_service_type, None)
             if obj.job_service_type
             else None,
-            # obj.job_service_type,
-            # nodes="all" if isinstance(obj.nodes, AllNodes) else None,
             nodes="all" if obj.nodes else None,
             status=obj.status,
             port=obj.port,
+            # ssh_public_keys=get_property(obj.properties, "sshPublicKeys"),
             properties=obj.properties,
         )
 
     @classmethod
-    def _from_rest_job_services(cls, services: Dict[str, RestJobService]) -> Dict[str, "JobService"]:
-        """Resolve Dict[str, RestJobService] to Dict[str, JobService]"""
+    def _from_rest_job_services(
+        cls, services: Dict[str, RestJobService]
+    ) -> Dict[
+        str, Union["JobService", "JupyterLabJobService", "SshJobService", "TensorBoardJobService", "VsCodeJobService"]
+    ]:
+        # """Resolve Dict[str, RestJobService] to Dict[str, Specific JobService]"""
         if services is None:
             return None
 
         result = {}
         for name, service in services.items():
-            result[name] = JobService._from_rest_object(service)
+            if service.job_service_type == JobServiceTypeNames.RestNames.JUPYTER_LAB:
+                result[name] = JupyterLabJobService._from_rest_object(service)
+            elif service.job_service_type == JobServiceTypeNames.RestNames.SSH:
+                result[name] = SshJobService._from_rest_object(service)
+            elif service.job_service_type == JobServiceTypeNames.RestNames.TENSOR_BOARD:
+                result[name] = TensorBoardJobService._from_rest_object(service)
+            elif service.job_service_type == JobServiceTypeNames.RestNames.VS_CODE:
+                result[name] = VsCodeJobService._from_rest_object(service)
+            else:
+                result[name] = JobService._from_rest_object(service)
         return result
+
+
+@experimental
+class JobService(JobServiceBase):
+    @classmethod
+    def _from_rest_object(cls, obj: RestJobService) -> "JobService":
+        return cls._from_rest_job_service_object(obj)
+
+    def _to_rest_object(self) -> RestJobService:
+        return self._to_rest_job_service()
+
+
+@experimental
+class SshJobService(JobServiceBase):
+    def __init__(
+        self,
+        *,
+        endpoint: Optional[str] = None,
+        job_service_type: Optional[Literal["ssh"]] = "ssh",
```

Comment:
We don't need to take type as an argument for this constructor.

Reply:
If I remove type in the constructor, schema mapping doesn't happen to these specific types and it gets mapped to JobService. I tried adding  `self.job_service_type = JobServiceTypeNames.EntityNames.SSH` inside def__init_ and it did not help

Reply:
Thanks @luigiw. data.pop() helped.